File: laojun-admin-web/src/services/system.ts
import { request } from './api';

// 后端系统设置模型（下划线风格）
interface BackendSystemSetting {
  id: string;
  key: string;
  value: string;
  type: 'string' | 'number' | 'boolean' | 'json';
  category: string;
  description?: string;
  is_public: boolean;
  is_sensitive?: boolean;
  created_at?: string;
  updated_at?: string;
}

// 前端系统配置模型（驼峰风格）
export interface SystemConfig {
  id: string;
  key: string;
  value: string;
  type: 'string' | 'number' | 'boolean' | 'json';
  category: string;
  description?: string;
  isPublic: boolean;
  isSensitive?: boolean;
  createdAt?: string;
  updatedAt?: string;
}

// 审计日志条目（与前端使用对齐）
export interface AuditLog {
  id: string;
  userId?: string;
  targetId?: string;
  targetType?: string;
  action: string;
  level: 'debug' | 'info' | 'warn' | 'error' | 'fatal';
  description?: string;
  oldData?: string;
  newData?: string;
  ipAddress?: string;
  userAgent?: string;
  createdAt: string;
}

// 性能指标
export interface Metrics {
  cpuUsage: number;
  memoryUsage: number;
  diskUsage: number;
}

const mapSetting = (s: BackendSystemSetting): SystemConfig => ({
  id: s.id,
  key: s.key,
  value: s.value,
  type: s.type,
  category: s.category,
  description: s.description,
  isPublic: !!s.is_public,
  isSensitive: !!s.is_sensitive,
  createdAt: s.created_at,
  updatedAt: s.updated_at,
});

export const systemService = {
  // 获取所有系统配置
  async getConfigs(): Promise<SystemConfig[]> {
    const res = await request.get<BackendSystemSetting[]>(`/system/configs`);
    return (res || []).map(mapSetting);
  },

  // 保存系统配置（批量）
  async saveConfigs(settings: SystemConfig[]): Promise<void> {
    // 将驼峰转换为后端下划线 & 保持 value 为字符串（JSON文本）
    const payload: BackendSystemSetting[] = settings.map(s => ({
      id: s.id,
      key: s.key,
      value: s.value,
      type: s.type,
      category: s.category,
      description: s.description,
      is_public: !!s.isPublic,
      is_sensitive: !!s.isSensitive,
      updated_at: s.updatedAt,
      created_at: s.createdAt,
    }));
    await request.post(`/system/configs`, payload);
  },

  // 获取审计日志（分页）
  async getLogs(params: { page?: number; pageSize?: number; level?: string; module?: string; startDate?: string; endDate?: string; }): Promise<{ data: AuditLog[]; total: number; }> {
    const query = {
      page: params.page ?? 1,
      pageSize: params.pageSize ?? 10,
      level: params.level,
      module: params.module,
      startDate: params.startDate,
      endDate: params.endDate,
    };
    const res = await request.get<{ data: any[]; total: number }>(`/system/logs`, { params: query });
    const logs: AuditLog[] = (res?.data || []).map((l: any) => ({
      id: l.id,
      userId: l.user_id,
      targetId: l.target_id,
      targetType: l.target_type,
      action: l.action,
      level: l.level,
      description: l.description,
      oldData: l.old_data,
      newData: l.new_data,
      ipAddress: l.ip_address,
      userAgent: l.user_agent,
      createdAt: l.created_at,
    }));
    return { data: logs, total: res?.total || 0 };
  },

  // 清理审计日志
  async clearLogs(params: { level?: string; module?: string; before?: string; }): Promise<number> {
    const res = await request.delete<{ affected: number }>(`/system/logs`, { params });
    return (res as any)?.affected ?? 0;
  },

  // 获取性能指标
  async getMetrics(): Promise<Metrics> {
    return await request.get<Metrics>(`/system/metrics`);
  }
};

File: laojun-admin-api/internal/models/system.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// SystemSetting 系统设置模型
type SystemSetting struct {
	ID          uuid.UUID `json:"id" db:"id"`
	Key         string    `json:"key" db:"key"`
	Value       string    `json:"value" db:"value"`    // JSON字符串，兼容多类型存储
	Type        string    `json:"type" db:"data_type"` // string, number, boolean, json
	Category    string    `json:"category" db:"category"`
	Description string    `json:"description" db:"description"`
	IsPublic    bool      `json:"is_public" db:"is_public"`
	IsSensitive bool      `json:"is_sensitive" db:"is_sensitive"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
}

// AuditLog 审计日志模型（查询返回）
type AuditLog struct {
	ID          uuid.UUID  `json:"id" db:"id"`
	UserID      uuid.UUID  `json:"user_id" db:"user_id"`
	TargetID    *uuid.UUID `json:"target_id" db:"target_id"`
	TargetType  string     `json:"target_type" db:"target_type"`
	Action      string     `json:"action" db:"action"`
	Level       string     `json:"level" db:"level"`
	Description string     `json:"description" db:"description"`
	OldData     string     `json:"old_data" db:"old_data"`
	NewData     string     `json:"new_data" db:"new_data"`
	IPAddress   string     `json:"ip_address" db:"ip_address"`
	UserAgent   string     `json:"user_agent" db:"user_agent"`
	CreatedAt   time.Time  `json:"created_at" db:"created_at"`
}

// Metrics 性能指标
type Metrics struct {
	CPUUsage    float64 `json:"cpuUsage"`
	MemoryUsage float64 `json:"memoryUsage"`
	DiskUsage   float64 `json:"diskUsage"`
}

// SystemInfo 系统信息
type SystemInfo struct {
	// 系统版本信息
	SystemName  string `json:"systemName"`
	Version     string `json:"version"`
	BuildTime   string `json:"buildTime"`
	GitCommit   string `json:"gitCommit"`
	Environment string `json:"environment"`

	// 运行环境信息
	OS           string `json:"os"`
	Architecture string `json:"architecture"`
	GoVersion    string `json:"goVersion"`
	NodeVersion  string `json:"nodeVersion"`
	Database     string `json:"database"`
	Redis        string `json:"redis"`
	Nginx        string `json:"nginx"`

	// 许可证信息
	License LicenseInfo `json:"license"`
}

// LicenseInfo 许可证信息
type LicenseInfo struct {
	Type        string `json:"type"`        // 许可证类型：commercial, open_source, trial
	Description string `json:"description"` // 许可证描述
	ExpiryDate  string `json:"expiryDate"`  // 有效期至
	IsValid     bool   `json:"isValid"`     // 是否有效
}
